spline-shadow/client-ui/src/app/store/reducers/attribute.reducer.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as _ from 'lodash';
+import { AttributeType } from '../../model/types/attributeType';
+import { AttributeVM } from '../../model/viewModels/attributeVM';
+import { CytoscapeGraphVM } from '../../model/viewModels/cytoscape/cytoscapeGraphVM';
+import { GraphNode } from '../../model/viewModels/cytoscape/graphNodeVM';
+import * as AttributesAction from '../actions/attributes.actions';
+
+export type Action = AttributesAction.AttributeActions
+
+export function attributeReducer(state: any, action: Action) {
+    switch (action.type) {
+        case AttributesAction.AttributesActionTypes.ATTRIBUTES_GET: return { ...state, ...buildAttributeGraph(action.payload) }
+        default: return null
+    }
+}
+
+export function getAttributeType(attribute: AttributeVM): string {
+    switch (attribute.dataType._type) {
+        case AttributeType.Struct: return '{ ... }'
+        case AttributeType.Array:
+            if (attribute.dataType.elementDataType.dataType.children != null) {
+                return '[{ ... }]'
+            }
+            return '[' + attribute.dataType.elementDataType.dataType.name + ']'
+        case AttributeType.Simple: return attribute.dataType.name
```

Comment:
Please always put 'return` on a new line, otherwise it's difficult to read.

Reply:
fixed